numba/parfors/array_analysis.py

```diff
@@ -450,10 +451,9 @@ def _get_names(self, obj):
             return tuple(self._get_names(x)[0] for x in obj)
         elif isinstance(obj, int):
             return (obj,)
-        else:
-            raise NotImplementedError(
-                "ShapeEquivSet does not support {}".format(obj)
-            )
+        warnings.warn(errors.NumbaPerformanceWarning(
```

Comment:
I think throwing a warning here may be too much, since these cases are not likely to affect performance. Maybe some debug/trace print is enough.